configtype/time.go

```diff
@@ -0,0 +1,164 @@
+package configtype
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/invopop/jsonschema"
+)
+
+type timeType int
+
+const (
+	timeTypeZero timeType = iota
+	timeTypeFixed
+	timeTypeRelative
+)
+
+// Time is a wrapper around time.Time that should be used in config
+// when a time type is required. We wrap the time.Time type so that
+// the spec can be extended in the future to support other types of times
+type Time struct {
+	typ      timeType
+	time     time.Time
+	duration Duration
+}
+
+func NewTime(t time.Time) Time {
```

Comment:
I would remove this in favor of `ParseTime` so there's only a single way to create this type